crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -0,0 +1,129 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind};
+use biome_console::markup;
+use biome_js_syntax::JsIdentifierBinding;
+use biome_rowan::AstNode;
+use regex::Regex;
+
+// List of sensitive patterns
+const SENSITIVE_PATTERNS: &[&str] = &[
+    r"xox[p|b|o|a]-[0-9]{12}-[0-9]{12}-[0-9]{12}-[a-z0-9]{32}",  // Slack Token
+    r"-----BEGIN RSA PRIVATE KEY-----",  // RSA Private Key
```

Comment:
I wonder if it wouldn't be faster to use String::contains for simple strings which do not require regex?

Reply:
Had it that way for simplicity, for parity with https://github.com/nickdeis/eslint-plugin-no-secrets/blob/master/regexes.js

Reply:
I will refactor this entire structure to match with [this comment](https://github.com/biomejs/biome/pull/3823#discussion_r1749236275).

Reply:
Done. Please check the latest code.